audit/schema.sql

```diff
@@ -0,0 +1,35 @@
+CREATE TABLE decrypt_event (
+  id SERIAL PRIMARY KEY,
+  timestamp TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+  username TEXT,
+  file TEXT
+);
+
+CREATE RULE decrypt_event_delete_protection AS ON DELETE TO decrypt_event DO INSTEAD NOTHING;
+
+CREATE TABLE encrypt_event (
+  id SERIAL PRIMARY KEY,
+  timestamp TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+  username TEXT,
+  file TEXT
+);
+
+CREATE RULE encrypt_event_delete_protection AS ON DELETE TO encrypt_event DO INSTEAD NOTHING;
+
+CREATE TABLE rotate_event (
+  id SERIAL PRIMARY KEY,
+  timestamp TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+  username TEXT,
+  file TEXT
+);
```

Comment:
All three tables have the same columns, so maybe they should just be one table with an `action` column added that reflects the decrypt, encrypt or rotate event?

Reply:
Doing that could cripple us when we want to add new event types (I expressed my opinion on this here: https://github.com/mozilla/sops/pull/256#discussion_r142147597)

Reply:
One could argue that changing the DB schema to add new type is more cumbersome that putting everything into a single table. I agree with the argument that we may want extra data, and maybe adding a `jsonb` column with `details` would help?

Ultimately, this should just be a blockchain! :trollface:

Reply:
Are these the only kind of audit events that will exist within sops? Events where it's "`action` on `file` by `user`?

Reply:
@jvehent @autrilla I'm impartial to this, but I would say that unless we expect the DB to have high throughput of writes migrating for either-or shouldn't be too difficult.

Reply:
I agree that migrating from either if it turns out the other way is better won't be difficult. @jvehent is right that changing the DB schema every time a new event type is added (hopefully this won't happen too often) is cumbersome, at least how we currently manage schema changes.